Mathlib/Data/Matroid/Minor/Basic.lean

```diff
@@ -0,0 +1,289 @@
+  rw [← restrict_compl, ← restrict_compl, ← restrict_compl, restrict_restrict_eq,
+    restrict_ground_eq, diff_diff]
+  simp [diff_subset]
+
+lemma delete_comm (M : Matroid α) (D₁ D₂ : Set α) : M ＼ D₁ ＼ D₂ = M ＼ D₂ ＼ D₁ := by
+  rw [delete_delete, union_comm, delete_delete]
+
+lemma delete_inter_ground_eq (M : Matroid α) (D : Set α) : M ＼ (D ∩ M.E) = M ＼ D := by
+  rw [← restrict_compl, ← restrict_compl, diff_inter_self_eq_diff]
+
+lemma delete_eq_delete_iff : M ＼ D₁ = M ＼ D₂ ↔ D₁ ∩ M.E = D₂ ∩ M.E := by
+  rw [← delete_inter_ground_eq, ← M.delete_inter_ground_eq D₂]
+  refine ⟨fun h ↦ ?_, fun h ↦ by rw [h]⟩
+  apply_fun (M.E \ Matroid.E ·) at h
+  simp_rw [delete_ground, diff_diff_cancel_left inter_subset_right] at h
+  assumption
+
+lemma IsRestriction.restrict_delete_of_disjoint (h : N ≤r M) (hX : Disjoint X N.E) :
+    N ≤r (M ＼ X) := by
+  obtain ⟨D, hD, rfl⟩ := isRestriction_iff_exists_eq_delete.1 h
+  refine isRestriction_iff_exists_eq_delete.2 ⟨D \ X, diff_subset_diff_left hD, ?_⟩
+  rwa [delete_delete, union_diff_self, union_comm, ← delete_delete, eq_comm,
+    delete_eq_self_iff]
+
+lemma IsRestriction.restriction_deleteElem (h : N ≤r M) (he : e ∉ N.E) : N ≤r M ＼ {e} :=
+  h.restrict_delete_of_disjoint (by simpa)
+
+@[simp]
+lemma delete_indep_iff : (M ＼ D).Indep I ↔ M.Indep I ∧ Disjoint I D := by
+  rw [← restrict_compl, restrict_indep_iff, subset_diff, ← and_assoc,
+    and_iff_left_of_imp Indep.subset_ground]
+
+lemma deleteElem_indep_iff : (M ＼ {e}).Indep I ↔ M.Indep I ∧ e ∉ I := by
+  simp
+
+lemma Indep.of_delete (h : (M ＼ D).Indep I) : M.Indep I :=
+  (delete_indep_iff.mp h).1
+
+lemma Indep.indep_delete_of_disjoint (h : M.Indep I) (hID : Disjoint I D) : (M ＼ D).Indep I :=
+  delete_indep_iff.mpr ⟨h, hID⟩
+
+lemma indep_iff_delete_of_disjoint (hID : Disjoint I D) : M.Indep I ↔ (M ＼ D).Indep I :=
+  ⟨fun h ↦ h.indep_delete_of_disjoint hID, fun h ↦ h.of_delete⟩
+
+@[simp]
+lemma delete_dep_iff : (M ＼ D).Dep X ↔ M.Dep X ∧ Disjoint X D := by
+  rw [dep_iff, dep_iff, delete_indep_iff, delete_ground, subset_diff]; tauto
+
+@[simp]
+lemma delete_isBase_iff : (M ＼ D).IsBase B ↔ M.IsBasis B (M.E \ D) := by
+  rw [← restrict_compl, isBase_restrict_iff]
+
+@[simp]
+lemma delete_isBasis_iff : (M ＼ D).IsBasis I X ↔ M.IsBasis I X ∧ Disjoint X D := by
+  rw [← restrict_compl, isBasis_restrict_iff, subset_diff, ← and_assoc,
+    and_iff_left_of_imp IsBasis.subset_ground]
+
+@[simp]
+lemma delete_isBasis'_iff : (M ＼ D).IsBasis' I X ↔ M.IsBasis' I (X \ D) := by
+  rw [isBasis'_iff_isBasis_inter_ground, delete_isBasis_iff, delete_ground, diff_eq, inter_comm M.E,
+    ← inter_assoc, ← diff_eq, ← isBasis'_iff_isBasis_inter_ground, and_iff_left_iff_imp,
+    inter_comm, ← inter_diff_assoc]
+  exact fun _ ↦ disjoint_sdiff_left
+
+lemma IsBasis.of_delete (h : (M ＼ D).IsBasis I X) : M.IsBasis I X :=
+  (delete_isBasis_iff.mp h).1
+
+lemma IsBasis.to_delete (h : M.IsBasis I X) (hX : Disjoint X D) : (M ＼ D).IsBasis I X := by
```

Comment:
I think this would be the naming convention:
```suggestion
lemma IsBasis.delete (h : M.IsBasis I X) (hX : Disjoint X D) : (M ＼ D).IsBasis I X := by
```